fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,435 @@
+        # automatically cleaning up the backend's index
+        brain_method.register_run(samples, brain_key, overwrite=False)
+
+    results = brain_method.initialize(samples, brain_key)
+    results.set_threshold(threshold)
+    leaks = results.leaks
+
+    if brain_key is not None:
+        brain_method.save_run_results(samples, brain_key, results)
+    if results._save_similarity_index:
+        results._similarity_method.save_run_results(
+            samples, similarity_brain_key, results._similarity_index
+        )
+
+    return results, leaks
+
+
+### GENERAL
+
+
+class LeakySplitsConfig(fob.BrainMethodConfig):
+    def __init__(
+        self,
+        split_views=None,
+        split_field=None,
+        split_tags=None,
+        similarity_brain_key=None,
+        embeddings_field=None,
+        model=None,
+        model_kwargs=None,
+        similarity_backend=None,
+        similarity_config_dict=None,
+        **kwargs,
+    ):
+        self.split_views = split_views
+        self.split_field = split_field
+        self.split_tags = split_tags
+        self.similarity_brain_key = similarity_brain_key
+        self.embeddings_field = embeddings_field
+        self.model = model
+        self.model_kwargs = model_kwargs
+        self.similarity_backend = similarity_backend
+        self.similarity_config_dict = similarity_config_dict
+        super().__init__(**kwargs)
+
+    @property
+    def type(self):
+        return "Data Leakage"
+
+    @property
+    def method(self):
+        return "Similarity"
+
+
+class LeakySplits(fob.BrainMethod):
+    def initialize(self, samples, brain_key=None):
+        return LeakySplitsIndex(samples, self.config, brain_key)
+
+
+class LeakySplitsIndex(fob.BrainResults):
+    def __init__(self, samples, config, brain_key):
+        super().__init__(samples, config, brain_key)
+
+        # process arguments to work only with views
+        self.split_views = _to_views(
+            samples,
+            self.config.split_views,
+            self.config.split_field,
+            self.config.split_tags,
+        )
+        self.id2split = self._id2splitConstructor()
+        self._leak_threshold = 0.2
+        self._last_computed_threshold = None
+        self._leaks = None
+
+        # similarity index setup
+        self._similarity_index = None
+        self._save_similarity_index = False
+        index_found = False
+        if self.config.similarity_brain_key is not None:
+            # Load similarity brain run if it exists
+            if (
+                self.config.similarity_brain_key
+                in self.samples._dataset.list_brain_runs()
+            ):
+                self._similarity_index = (
+                    self.samples._dataset.load_brain_results(
+                        self.config.similarity_brain_key, load_view=True
+                    )
+                )
+                if self._similarity_index is not None:
```

Comment:
is this condition ever met? If so, when?

Reply:
was supposed to be in case of bad loading, I can remove it.

Reply:
honestly I'm not sure what would be returned in that case anyways